prow/cmd/pipeline/controller.go

```diff
@@ -690,6 +690,16 @@ func updateProwJobState(c reconciler, pj *prowjobv1.ProwJob, state prowjobv1.Pro
 			}
 			npj.Status.State = state
 			npj.Status.Description = msg
+
+			// Set the status build ID and URL if they're empty
+			for _, r := range runs {
+				if npj.Status.BuildID != "" && npj.Status.URL != "" {
+					continue
+				}
+				logrus.Infof("Setting build ID and URL for ProwJob/%s from PipelineRun %s", pj.GetName(), r.Name)
+				npj.Status.BuildID = getBuildNumber(r)
```

Comment:
Isn't this going to set the build number of metapipline in case is the first run? Shouldn't be instead the build number of the effective pipeline?

Reply:
The metapipeline and effective pipeline always have the same build number.

Reply:
great!